taichi/util/hash.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <functional>
+#include <stddef.h>
+
+namespace std {
+
+namespace {
+inline void hash_combine(size_t &seed, size_t value) {
+  seed ^= value + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+}
+}  // namespace
+
+template <typename T>
+struct hash<std::vector<T>> {
```

Comment:
This is quite dangerous, don't think we should be extending std namespace with a template that's as wide as this. We should move this into a custom hasher